client/dashboard/me/notifications-extras/extras-toggle-card/test/index.test.tsx

```diff
@@ -0,0 +1,239 @@
+			community: true,
+			reports: true,
+		};
+
+		renderExtrasToggleCard( { extraSettings } );
+
+		expect( screen.getByLabelText( 'Suggestions' ) ).toBeChecked();
+		expect( screen.getByLabelText( 'Research' ) ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Community' ) ).toBeChecked();
+		expect( screen.getByLabelText( 'Promotions' ) ).not.toBeChecked();
+		expect( screen.getByLabelText( 'Reports' ) ).toBeChecked();
+	} );
+
+	it( 'calls onMutate with only changed item when individual toggle is changed', async () => {
+		renderExtrasToggleCard();
+
+		const marketingToggle = screen.getByLabelText( 'Suggestions' );
+		await userEvent.click( marketingToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( { marketing: true } );
+	} );
+
+	it( 'calls onMutate when "Subscribe to all" is clicked', async () => {
+		renderExtrasToggleCard();
+
+		const subscribeAllToggle = screen.getByLabelText( 'Subscribe to all' );
+		await userEvent.click( subscribeAllToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( {
+			marketing: true,
+			research: true,
+			community: true,
+			promotion: true,
+			news: true,
+			digest: true,
+			reports: true,
+			news_developer: true,
+			scheduled_updates: true,
+		} );
+	} );
+
+	it( 'calls onMutate with only changed values when "Unsubscribe from all" is clicked', async () => {
+		const extraSettings = {
+			...defaultExtraSettings,
+			marketing: true,
+			research: true,
+			community: true,
+		};
+
+		renderExtrasToggleCard( { extraSettings } );
+
+		const unsubscribeAllToggle = screen.getByLabelText( 'Unsubscribe from all' );
+		await userEvent.click( unsubscribeAllToggle );
+
+		expect( mockOnMutate ).toHaveBeenCalledWith( {
+			marketing: false,
+			research: false,
+			community: false,
+		} );
+	} );
+
+	it( 'disables all toggles when isSaving is true', () => {
+		renderExtrasToggleCard( { isSaving: true } );
+
+		expect( screen.getByLabelText( 'Subscribe to all' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Suggestions' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Research' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Community' ) ).toBeDisabled();
+	} );
+
+	it( 'disables all toggles when extraSettings is undefined', () => {
+		renderExtrasToggleCard( { extraSettings: undefined } );
+
+		expect( screen.getByLabelText( 'Subscribe to all' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Suggestions' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Research' ) ).toBeDisabled();
+		expect( screen.getByLabelText( 'Community' ) ).toBeDisabled();
+	} );
+
+	it( 'does not call onMutate when extraSettings is undefined', async () => {
+		renderExtrasToggleCard( { extraSettings: undefined } );
+
+		const subscribeAllToggle = screen.getByLabelText( 'Subscribe to all' );
+		await userEvent.click( subscribeAllToggle );
+
+		expect( mockOnMutate ).not.toHaveBeenCalled();
+	} );
+
+	it( 'handles individual toggle changes correctly', async () => {
```

Comment:
It seems duplicated with this case "calls onMutate with only changed item when individual toggle is changed"  line 120, maybe we can delete one of them.